queries/play.py

```diff
@@ -0,0 +1,372 @@
+
+
+def find_youtube_playlists(q: str, limit: int = 3) -> List[str]:
+    """Find URLs for playlists given a search string via the YouTube API."""
+    vids = []
+    try:
+        r = search_youtube(q, types=["playlist"], limit=limit)
+        if r is None or r.items is None:
+            return vids
+
+        for i in r.items:
+            item = i.to_dict()
+            if "id" not in item or "playlistId" not in item["id"]:
+                continue
+            playlist_id = item["id"]["playlistId"]
+            pl_vids = YT_API.get_playlist_items(playlist_id=playlist_id, count=1)
+            if not pl_vids.items:
+                continue
+            first_vid_id = pl_vids.items[0].snippet.resourceId.videoId
+            vids.append(_YOUTUBE_PLAYLIST_URL.format(first_vid_id, playlist_id))
+    except Exception as e:
+        logging.error(f"Error communicating with YouTube API: {e}")
+
+    return vids
+
+
+def rand_yt_playlist_for_genre(
+    genre_name: str, limit: int = 5, fallback=None
+) -> Optional[str]:
+    """Given a musical genre name, search for YouTube playlists and return a
+    URL to a randomly selected one, with an (optional) fallback video URL."""
+    urls = find_youtube_playlists(genre_name, limit=limit)
+    if urls:
+        return choice(urls)
+    return fallback
+
+
+# Musical genres
+def _play_jazz(qs: str, q: Query, matches: Optional[Match[str]]) -> None:
+    # Caravan - Duke Ellington classic
+    fb = "https://www.youtube.com/watch?v=E5loTx0_KDE"
+    q.set_url(rand_yt_playlist_for_genre("jazz", fallback=fb))
+    q.set_key(matches.group(1) if matches else "")
+
+
+def _play_blues(qs: str, q: Query, matches: Optional[Match[str]]) -> None:
+    # How Long Blues - Jimmy & Mama Yancey
+    fb = "https://www.youtube.com/watch?v=jw9tMRhKEak"
+    q.set_url(rand_yt_playlist_for_genre("blues", fallback=fb))
+    q.set_key(matches.group(1) if matches else "")
+
+
+def _play_rock(qs: str, q: Query, matches: Optional[Match[str]]) -> None:
+    # Led Zeppelin - Immigrant Song
+    fb = "https://www.youtube.com/watch?v=y8OtzJtp-EM"
+    q.set_url(rand_yt_playlist_for_genre("classic rock", fallback=fb))
+    q.set_key(matches.group(1) if matches else "")
+
+
+def _play_classical(qs: str, q: Query, matches: Optional[Match[str]]) -> None:
+    # Beethoven - 9th symphony, 2nd movement
+    fb = "https://www.youtube.com/watch?v=iwIvS4yIThU"
+    q.set_url(rand_yt_playlist_for_genre("classical music", fallback=fb))
+    q.set_key(matches.group(1) if matches else "")
+
+
+def _play_electronic(qs: str, q: Query, matches: Optional[Match[str]]) -> None:
+    # Orbital - The Box
+    fb = "https://www.youtube.com/watch?v=qddG0iUSax4"
+    q.set_url(rand_yt_playlist_for_genre("retro electronic music", fallback=fb))
+    q.set_key(matches.group(1) if matches else "")
+
+
+# Play music from a randomly selected genre
+def _play_music(qs: str, q: Query, matches: Match[str]) -> None:
```

Comment:
matches parameter is not used, is that intentional? Also, probably should be Optional[Match[str]] if present.